File: Data Structures & Algorithms/combination-target-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    combinationSum(nums, target) {
        let res = []
        if (nums == null) {
            return res
        }
        nums.sort()
        const generateCombinationSum = (index,cur,total) => {
            if (total == target) {
                res.push([...cur])
                return
            }

            if (total > target || index >= nums.length) {
                return
            }
            cur.push(nums[index])
            generateCombinationSum(index,cur,total+nums[index])
            cur.pop()
            generateCombinationSum(index+1,cur,total)
        }

        generateCombinationSum(0,[],0)
        return res

    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = []
        if nums is None:
            return res
        
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                   return [i,j]
        return res
        